apis/management.cattle.io/v3/authn_types.go

```diff
@@ -162,6 +162,27 @@ type GithubConfigApplyInput struct {
 	Enabled      bool         `json:"enabled,omitempty"`
 }
 
+type GoogleOAuthConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthConfig        `json:",inline" mapstructure:",squash"`
+
+	ClientCredentialsFile string `json:"clientCredentialsFile,omitempty" norman:"required"`
```

Comment:
Singular, not a file, should show up as `type: password`.  `googleKuberentesEngineConfig` just uses `credential`.

I don't remember exactly what the two different sets of credentials were for other than one was optional to make groups work, but these probably need better names.  Who's the client?  Where do I get `ClientCredentialsFile` and `ServiceAccountFile` from?

Reply:
<img width="1440" alt="Screen Shot 2019-05-28 at 12 33 14 PM" src="https://user-images.githubusercontent.com/10672776/58508953-35b30b00-814a-11e9-8354-0c8939310253.png">
The client credentials file has the client ID and secret generated through the `Create Credentials` -> `OAuth Client ID` option above.
Once created, the download button on the right most side of a client id downloads a json file titled client_secret_<clientID>.json (That's why I named it ClientCredentialsFile)
So we can copy paste contents of this file/upload it directly as we do for saml providers’ cert/key/metadata. The golang-google-oauth library we use also accepts this json file when generating access token.
(Ref: https://developers.google.com/identity/protocols/OpenIDConnect)

<img width="1438" alt="Screen Shot 2019-05-28 at 1 14 41 PM" src="https://user-images.githubusercontent.com/10672776/58509139-b07c2600-814a-11e9-908f-e45b777b1223.png">
According to google docs this is how you can create a service account for domain delegation. And during creation it asks you to download a file containing a private key for the service account either in JSON or P12 format, and JSON is the recommended one. That's why I named this ServiceAccountFile
(Ref: https://developers.google.com/admin-sdk/directory/v1/guides/delegation)

Is this okay for the two files? or you think we should go with something else?

Yes I will change the type to password for both these, but I hadn't done this right now, since I'm using api ui to do all testing, and it becomes easier to input plaintext, so I was going to change it to password before it's merged

Reply:
Yes;  there are no files in an API, so `File` implies that you're looking for a path to a file, which you are not.

I would call them `oauthCredential` and `serviceAccountCredential`, since they're both the same "credential" format.